src/validation.cpp

```diff
@@ -2542,7 +2542,7 @@ bool static DisconnectTip(CValidationState& state, const CChainParams& chainpara
     UpdateTip(pindexDelete->pprev, chainparams);
     // Let wallets know transactions went from 1-confirmed to
     // 0-confirmed or conflicted:
-    GetMainSignals().BlockDisconnected(pblock, pindexDelete->pprev);
+    GetMainSignals().BlockDisconnected(pblock, pindexDelete);
```

Comment:
Is it actually safe to pass a local copy of a pointer here? Probably would be better to pass a shared_ptr instead, smth like 5ba050842bccc0cb1e79a14601dc3ee0d7850746.

Reply:
Should be perfectly safe. The `shared_ptr` is not needed here as we currently can assume that block indexes are never deleted (they are owned by `mapBlockIndex`), including the indexes referred by `pprev`. `shared_ptr` is usually needed for `CBlock` and `CTransaction` as there is no clear ownership or life-cycle defined.